opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp

```diff
@@ -81,6 +99,57 @@ namespace Opm {
                 for (size_t globalIndex = 0; globalIndex < multpvData.size(); globalIndex++)
                     values[globalIndex] *= multpvData[globalIndex];
             }
+
+            // deal with the region multiplier for porosity
+            if (deck->hasKeyword("MULTREGP")) {
+                const DeckKeyword& multregpKeyword = deck->getKeyword("MULTREGP");
+                for (int recordIdx = 0; recordIdx < multregpKeyword.size(); ++recordIdx) {
+                    const DeckRecord& multregpRecord = multregpKeyword.getRecord(recordIdx);
+
+                    int regionIdx = multregpRecord.getItem("REGION").template get<int>(0);
+                    std::string regionType = multregpRecord.getItem("REGION_TYPE").template get<std::string>(0);
+                    double multValue = multregpRecord.getItem("MULTIPLIER").template get<double>(0);
+                    boost::to_upper(regionType);
+
+                    // deal with the "convenience" feature of ECL: if the region index is
+                    // zero or negative, the record is ignored. (why are such lines not
+                    // simply commented out? so that mere mortals do not understand
+                    // what's going on?)
+                    if (regionIdx <= 0)
+                        continue;
+
+                    // implement a (documented) ECL bug: the region index must be unique
```

Comment:
for this particular issue, the code can hardly be shared because MULTREGT has two regions here,  while MULTREGP only has one. I haven't looked at MULTREGT too much, though, so I'm not even sure wheter the (sourceRegion, targetRegion) pairs must be unique. (Also, technically it would obviously be possible to do this using the same code path, but as far as I can see the abstraction layer required to hide the difference between a pair of integers and a single integer would be much larger than the code itself, so it would IMO be rather like an "obfuscation layer".)